test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```
@@ -0,0 +1,400 @@
+
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $badCert.Thumbprint
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $ers[0].FullyQualifiedErrorId | Should Be "CertificateCannotBeUsedForEncryption"
+    }
+
+    It "Verify no error when encrypting to wildcarded identifier for decryption" {
+        $ers = $null
+        $recipient = [System.Management.Automation.CmsMessageRecipient] "SomeCertificateThatDoesNotExist*"
+        $recipient.Resolve($ExecutionContext.SessionState, "Decryption", [ref] $ers)
+
+        $ers | Should Be $null
+        $recipient.Certificates.Count | Should Be 0
+    }
+
+    It "Verify Protect-CmsMessage emits recipient errors" {
+        try {
+            "Hello World" | Protect-CmsMessage -To "SomeThumbprintThatDoesNotExist" -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.ProtectCmsMessageCommand"
+        }
+    }
+
+    It "Verify CmsMessage cmdlets works with paths" {
+        try {
+            $randomNum = Get-Random -Minimum 1000 -Maximum 9999
+            $tempPath = Join-Path $TestDrive "$randomNum-Path-Test-File"
+            $encryptedPath = $tempPath + ".encrypted.txt"
+            "Hello World","How are you?" | Set-Content $tempPath
+
+            Protect-CmsMessage -Path $tempPath -To $certLocation -OutFile $encryptedPath
+
+            $message = Get-CmsMessage -LiteralPath $encryptedPath
+            $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+            $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+            $decrypted | Should Be "Hello World`r`nHow are you?`r`n"
+
+            $decrypted = Unprotect-CmsMessage -Path $encryptedPath -To $certLocation
+            $decrypted | Should Be "Hello World`r`nHow are you?`r`n"
+        } finally {
```

Without `Catch` can we silently skip the test if there is a throw?